src/UI/TransactionsToastList/components/TransactionToast/hooks/useTransactionToast.ts

```diff
@@ -85,25 +96,49 @@ export const useTransactionToast = ({
   };
 
   useEffect(() => {
-    if (!isCompleted || !lifetimeAfterSuccess || timeoutRef.current) {
+    if (
+      !isCompleted ||
+      !lifetimeAfterSuccess ||
+      lifetimeAfterSuccessTimeoutRef.current
+    ) {
       return;
     }
 
-    timeoutRef.current = setTimeout(() => {
+    lifetimeAfterSuccessTimeoutRef.current = setTimeout(() => {
       handleDeleteToast();
     }, lifetimeAfterSuccess);
 
     return () => {
-      if (timeoutRef.current) {
+      if (lifetimeAfterSuccessTimeoutRef.current) {
         // Clear timer on unmount and also delete the toast
         // The toast may have been removed before the timer finished by the re-rendering
         // of the toasts list during another toast removal from the store
         handleDeleteToast();
-        clearTimeout(timeoutRef.current);
+        clearTimeout(lifetimeAfterSuccessTimeoutRef.current);
       }
     };
   }, [lifetimeAfterSuccess, isCompleted]);
 
+  // Prevent the toast from infinite pending in case the API fails
+  // Mark the toast/transaction as timed out after specific seconds
+  useEffect(() => {
+    if (!isPending || lifetimeRef.current) {
+      return;
+    }
+
+    lifetimeRef.current = setTimeout(() => {
+      if (isPending) {
+        setIsToastTimedOut(true);
+      }
+    }, toastLifetime);
+
+    return () => {
+      if (lifetimeRef.current) {
+        clearTimeout(lifetimeRef.current);
```

Comment:
I think would be more correct to check the transaction on the server then close it instead of closing it by force